MaterialLib/MPL/Properties/CapillaryPressureSaturation/CapillaryPressureVanGenuchten.cpp

```diff
@@ -20,19 +18,21 @@ namespace MaterialPropertyLib
 {
 CapillaryPressureVanGenuchten::CapillaryPressureVanGenuchten(
     double const residual_liquid_saturation,
-    double const maximum_liquid_saturation, double const exponent,
-    double const p_b, double const max_capillary_pressure)
-    : _residual_saturation(residual_liquid_saturation),
-      _maximuml_saturation(maximum_liquid_saturation),
+    double const residual_gas_saturation,
+    double const exponent,
+    double const p_b,
+    double const maximum_capillary_pressure)
+    : _S_L_res(residual_liquid_saturation),
```

Comment:
An underscore followed immediately by an uppercase letter should be avoid. This is why I did not use this kind of naming as that in `SaturationVanGenuchten`
See:
https://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier

Reply:
> An underscore followed immediately by an uppercase letter should be avoided. This is why I did not use this kind of naming as that in `SaturationVanGenuchten`
This implementation uses same convention as the `SaturationVanGenuchten` [implementation](https://github.com/ufz/ogs/blob/master/MaterialLib/MPL/Properties/CapillaryPressureSaturation/SaturationVanGenuchten.h#L85-L88) for consistency.

> https://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier
The reference is interesting indeed. The rules should be checked automatically across the whole code base. Any ideas on how to do it?

Reply:
And again, if this affects multiple places, which it does, open a new issue/pull-request.

Reply:
As far as I understand this the rule has effect on the global scope only: [n3242 / C++11 draft](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2011/n3242.pdf) 17.6.4.3.2 [global names].

Reply:
You can keep `_S...` naming, and we can resolve this issue later (whether change or not). I just remembered  that there is case of `_A`  name caused some problem. Maybe the new C++ standard removed some identification reservation.

BTW, cpplint may be used to check the naming issue. @bilke  may know this.

Reply:
I still see the following statement on page 20 of [C++17 final draft](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/n4659.pdf). It is the same as before.
```
In addition, some identifiers are reserved for use by C++implementations
 and shall not be used otherwise; no diagnostic is required.
—(3.1)Each identifier that contains a double underscore__or begins
 with an underscore followed by an uppercase letter is reserved to
 the implementation
 for any use.
```

Reply:
I see. Google's style guide also recommends to put the underscores after the variable names.
Please open an issue. Intermediate solution suggestion is to keep the leading for consistent names. New, independent code parts should have trailing underscores for private class members. 
@wenqing Please find a way to automatically check this, preferably with clang-tidy or cppcheck, such that we can eliminate this issue step by step.